src/napari_omero/plugins/loaders.py

```diff
@@ -128,3 +133,72 @@ def get_lazy_plane(zct: tuple[int, ...]):
             c_stacks.append(da.stack(z_stack))
         t_stacks.append(da.stack(c_stacks))
     return da.stack(t_stacks)
+
+
+def get_pyramid_lazy(image: ImageWrapper) -> list[da.Array]:
+    """Get a pyramid of rgb dask arrays, loading tiles from OMERO."""
+    size_z = image.getSizeZ()
+    size_t = image.getSizeT()
+    size_c = image.getSizeC()
+    pixels = image.getPrimaryPixels()
+    dtype = PIXEL_TYPES.get(pixels.getPixelsType().value, None)
+
+    image._prepareRenderingEngine()
+    tile_w, tile_h = image._re.getTileSize()
+
+    def get_tile(tile_name):
+        """tile_name is 'level,z,t,x,y,w,h'."""
+        level, z, c, t, x, y, w, h = (int(n) for n in tile_name.split(","))
+        pix = image._conn.c.sf.createRawPixelsStore()
+        pix_id = image.getPixelsId()
+        try:
+            pix.setPixelsId(pix_id, False)
+            pix.setResolutionLevel(level)
+            tile = pix.getTile(z, c, t, x, y, w, h)
+            tile = np.frombuffer(tile, dtype=np.uint8)
+            tile = tile.reshape((h, w))
+            return tile
+        finally:
+            pix.close()
```

Comment:
Does a context manager make sense for this section?

Reply:
@psobolewskiPhD I think it does make sense to ensure that the PixelStore is closed for sure, I wouldn't know how it would affect the retrieval of the next item if it weren't. Where I am concerned this can stay 👍 

You *could* phrase it a a bit more explicit, if anything - for me it wasn't directly clear that I was looking at a context manager ^^ Something like this:

```python
from contextlib import contextmanager

@contextmanager
def raw_pixels_store(image):
    pix = image._conn.c.sf.createRawPixelsStore()
    try:
        yield pix
    finally:
        pix.close()
```

and then 

```python
with with raw_pixels_store(image) as pix:
  ...
```

Reply:
@will-moore do you have any thoughts on using a context manager here? I think the suggestion from Johannes is essentially what I was considering.

Reply:
Failing to call `pix.close()` shouldn't affect the next usage since with `pix = image._conn.c.sf.createRawPixelsStore()` we are creating a new store each time (instead of simply `conn.createRawPixelsStore()` which possibly reuses them).

But if you don't close the store then the server keeps it open which uses resources, file-handles etc.

I don't have a strong opinion on context manager, but if it helps clarify the code then 👍